fuse/ipns/ipns_unix.go

```diff
@@ -370,13 +384,21 @@ func (dir *Directory) Mkdir(ctx context.Context, req *fuse.MkdirRequest) (fs.Nod
 
 func (fi *File) Open(ctx context.Context, req *fuse.OpenRequest, resp *fuse.OpenResponse) (fs.Handle, error) {
 	if req.Flags&fuse.OpenTruncate != 0 {
-		log.Warning("Need to truncate file!")
+		log.Info("Need to truncate file!")
 		err := fi.fi.Truncate(0)
 		if err != nil {
 			return nil, err
 		}
 	} else if req.Flags&fuse.OpenAppend != 0 {
-		log.Warning("Need to append to file!")
+		log.Info("Need to append to file!")
+
+		// seek(0) essentially resets the file object, this is required for appends to work
+		// properly
+		_, err := fi.fi.Seek(0, os.SEEK_SET)
```

Comment:
Aall the actual reads and writes really are `ReadAt` / `WriteAt` style. Kernel keeps file offsets, doesn't even communicate them through FUSE. You seem to be doing that right.

The real thing you have to worry about is reads and writes happening at the same time. How does your `fi.fi` there like two Write calls at the same time?

Separate Handles is mostly useful for e.g. keeping track whether the handle was opened O_RDONLY or not, for example if your backing store is cheaper to open read-only than read-write.

The OpenTruncate logic there belongs in Setattr, the kernel will tell you when it wants to change the length of the file.